@here/harp-mapview/lib/Utils.ts

```diff
@@ -446,6 +460,229 @@ export namespace MapViewUtils {
+    ) {
+        if (
+            material instanceof THREE.RawShaderMaterial ||
+            material instanceof THREE.ShaderMaterial
+        ) {
+            const rawMaterial = material;
+            for (const name in rawMaterial.uniforms) {
+                if (rawMaterial.uniforms[name] !== undefined) {
+                    const uniform = rawMaterial.uniforms[name];
+                    if (uniform instanceof THREE.Texture) {
+                        estimateTextureSize(uniform, isVisible, objectSize);
+                    }
+                }
+            }
+        } else if (
+            material instanceof THREE.MeshBasicMaterial ||
+            material instanceof MapMeshBasicMaterial
+        ) {
+            const meshMaterial = material;
+            estimateTextureSize(meshMaterial.map, isVisible, objectSize);
+            estimateTextureSize(meshMaterial.aoMap, isVisible, objectSize);
+            estimateTextureSize(meshMaterial.specularMap, isVisible, objectSize);
+            estimateTextureSize(meshMaterial.alphaMap, isVisible, objectSize);
+            estimateTextureSize(meshMaterial.envMap, isVisible, objectSize);
+        } else if (material instanceof MapMeshStandardMaterial) {
+            const standardMaterial = material;
+            estimateTextureSize(standardMaterial.map, isVisible, objectSize);
+            estimateTextureSize(standardMaterial.aoMap, isVisible, objectSize);
+            estimateTextureSize(standardMaterial.alphaMap, isVisible, objectSize);
+            estimateTextureSize(standardMaterial.envMap, isVisible, objectSize);
+        } else if (
+            material instanceof THREE.LineBasicMaterial ||
+            material instanceof THREE.LineDashedMaterial
+        ) {
+            // Nothing to be done here
+        } else {
+            logger.warn("estimateMeshSize: unidentified material: ", material);
+        }
+    }
+
+    function estimateAttributeSize(
+        attribute: any,
+        attrName: string,
+        isVisible: boolean,
+        objectSize: MemoryUsage
+    ) {
+        let bytesPerElement = 4;
+        let attrBytes = 0;
+        if (attribute.array.BYTES_PER_ELEMENT !== undefined) {
+            bytesPerElement = attribute.array.BYTES_PER_ELEMENT;
+        }
+        if (attribute instanceof THREE.InterleavedBufferAttribute) {
+            const interAttr = attribute;
+            if (interAttr.offset !== 0) {
+                // Array is shared between attributes, count it only once
+                return;
+            }
+            attrBytes = bytesPerElement * attribute.count * attribute.itemSize;
+        } else if (attribute instanceof THREE.BufferAttribute) {
+            const bufferAttr = attribute;
+            attrBytes = bytesPerElement * bufferAttr.count * bufferAttr.itemSize;
+        } else {
+            logger.warn("estimateMeshSize: unidentified attribute: ", attrName);
+        }
+
+        objectSize.heapSize += attrBytes;
+        if (isVisible) {
```

Comment:
Same as above, even if it's not visible, resources will be resident on the GPU.

Reply:
Removed the "isVisible" test, because the handling of shared resources (like THREE.Materials and THREE.Attributes) does not support handling the visibility state of the resource.